core/vm/instructions.go

```diff
@@ -640,6 +653,10 @@ func opCreate2(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]
 }
 
 func opCall(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
+	if interpreter.readOnly && !value.IsZero() {
+		return nil, ErrWriteProtection
+	}
+
```

Comment:
I didn't remark on this particular one, because here it was needed -- you get `value` from the stack

Reply:
You're right, it was too early in the morning. Good that we have reviews 😅

Reply:
> I didn't remark on this particular one, because here it was needed -- you get `value` from the stack